### PR TITLE
forgejo: 8.0.3 -> 9.0.0

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Things done

Upgrades forgejo to v9.
Forgejo v9 released on 16th October.
See https://codeberg.org/forgejo/forgejo/issues/5380
Release Notes: https://codeberg.org/forgejo/forgejo/milestone/7235

Also added myself as a maintainer :)

- [x] Use buildGoModule again once #342429 reached master - [PR Tracker](https://nixpk.gs/pr-tracker.html?pr=342429)
- [x] Wait for release on 16th October

---

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -42,6 +42,8 @@
   Users can use it by `services.displayManager.ly.enable` and config it by
   `services.displayManager.ly.settings` to generate `/etc/ly/config.ini`
 
+- `forgejo` has been upgraded to version 9.0, see the [release notes](https://codeberg.org/forgejo/forgejo/milestone/7235).
```
Comment (emilylange) [inline nixos/doc/manual/release-notes/rl-2411.section.md:45]: I lean towards moving that release note entry out of the "_Highlights_" section into the "_Backward Incompatibilities_" further down below where we already have two entries for Forgejo.
So all of them are side by side.

Well and jumping from v7 (`forgejo` in `24.05`) to v9 are technically two major updates with breakages.

But let me know what you think -- open to discuss :)

We could also use that opportunity to mention `forgejo-lts` somewhere. Which is something we wanted to do a while ago but postponed so far.

Comment (adamcstephens) [reply nixos/doc/manual/release-notes/rl-2411.section.md:45]: I agree it's probably worth moving to the Backward Incompat section.

Maybe something like

> `services.foregejo.package` now defaults to `forgejo-lts`, the Long Term Support version of Forgejo. The latest version can be selected by setting `services.forejo.package = pkgs.forgejo`, with the understanding that this package could have breaking changes during the life of a stable NixOS release.

Comment (NyCodeGHG) [reply nixos/doc/manual/release-notes/rl-2411.section.md:45]: @emilylange 
> I lean towards moving that release note entry out of the "_Highlights_" section into the "_Backward Incompatibilities_" further down below where we already have two entries for Forgejo.
> So all of them are side by side.

sounds like a good idea.

> Well and jumping from v7 (forgejo in 24.05) to v9 are technically two major updates with breakages.

Right, we should mention that as well.

@adamcstephens 
> with the understanding that this package could have breaking changes during the life of a stable NixOS release.

Wouldn't it be better to introduce e.g. `forgejo_9`, so it's a bit more explicit to upgrade? Not sure how to handle this after EOL, just a thought.

Comment (adamcstephens) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: This was discussed in https://github.com/NixOS/nixpkgs/pull/331187 but there are only ever two releases of forgejo.

Comment (adamcstephens) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: If a use has changed the default from the LTS version, I personally consider that explicit opting in.

**Comment by adamcstephens** [comment]
This needs to be moved to the release version so we can get it in. There is no overlap of support with v8.

File: nixos/doc/manual/release-notes/rl-2411.section.md (hunk 1 above)
Comment (adamcstephens) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: If we want to revisit the naming decision again, let's do so in another PR.

**Review by jalil-salame — APPROVED** [review]
Forgejo v9 was released, the PR can be marked as ready now c:

**Comment by NyCodeGHG** [comment]
Rebased onto master to fix the conflict and updated to the release version!

**Review by adamcstephens — APPROVED** [review]

**Review by emilylange — APPROVED** [review]
The changes upstream made to the `Makefile` do not apply to us or rather don't require actions from us.

And the new journald logging is detected automatically, so no need to change our module defaults.

LGTM, thanks.

**Pull request merged**